docs/framework/contributing/code-style.md

```diff
@@ -1062,3 +1064,49 @@ import ClassicEditor from '@ckeditor/ckeditor5-build-classic/src/ckeditor';
 ```
 
 [History of the change.](https://github.com/ckeditor/ckeditor5/issues/13689)
+
+### Declaring module augmentation for the core package: `allow-declare-module-only-in-augmentation-file`
+
+<info-box warning>
+  This rule should only be used on `.ts` files.
+</info-box>
+
+The main entry points (`index.ts` files) in most modules have a side effect import `import './augmentation'` which uses module augmentation to populate the editor types with information about available plugins, configurations and commands.
+
+This rule forces all `declare module '@ckeditor/ckeditor5-core'` to be defined in this `augmentation.ts` file.
+
+### Importing from modules: `allow-imports-only-from-main-package-entry-point`
```

Comment:
There are some code examples in this file in "DLL Builds: ckeditor5-rules/ckeditor-imports" section, that will be invalid under this rule. Please review all code samples in `code-style.md`.